### ISSUE TITLE
[a11y]: File Uploader - "Drag & drop files or click to upload" announced multiple times

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16329]
### Package

@carbon/react

### Browser

Firefox

### Package version

v1.55.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

"Drag & drop files or click to upload" label is announced 3 times on arrow key navigation in which button can be activated only on first announcement

### Recommended Fix
Ensure that the label is announced only once

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--drag-and-drop-upload-container-example-application

### Steps to reproduce

none

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
I believe this is related to an issue that Tom mentions in the [Carbon Accessibility Guild slack channel](https://ibm-studios.slack.com/archives/G01GCBCGTPV/p1738166582396489).

> Question on the FileUploader...  At the moment, screen reader users hear a lot of duplication - they hear the Carbon button, they hear the native file uploader button, and then the label again with the file selected. I'm assuming this is visually hidden rather than display none for some javascript reason since as far as I can tell, the screen reader users are still interacting with the visual button. Can we add aria-hidden to the label and to the input so that it's "visually hidden" from the screen readers also?
